ronan_solutions/most_common_element_impl.py

```diff
@@ -8,6 +9,28 @@ def most_common_element(elements: List[Any]) -> List[Any]:
     there are three 0s, three 3s, two 2s, one 1 and one 4, so the correct solution is [0, 3]
     or [3, 0].
     """
-    return [None]
+    #I created a dictionary because I wanted to store not only the number of times a value appears, 
+    # but also the value itself at the same time
+    countPerItem = {}
+    #iteration over the list
+    for i in List:
+            #count the number of times a number appears in the list and attach that as the key to the value that reflects the 
+            #original item
+            countPerItem.append(i,list.count(i))
+        
+    #sort through the dictionary to find the item(s) with the largest value
+    k=0
+    for j in countPerItem.values():
+        if j > k: 
+            k=j
+    
```

Comment:
Here's another issue: now that you have the count, how do you find the max? This way will work, but you can avoid having to do another entire traversal by simply keeping track of the max as you go. (Think about how you might be able to do this during the original traversal.)

Reply:
Also: you shouldn't have to do the third traversal below. Instead, keep track of both the max count and all keys that match it:

```python
    max = None   # start with None, because the value doesn't mean anything yet (also use meaningful names)
    eles = []         # lets make this  a list because we want to keep track of all elements
    for ele in countPerItem.keys():
        if countPerItem[ele] == max:
            eles.append(ele)
        elif countPerItem[ele] > max:
            max = countPerItem[ele]
            eles = [ele]
       #no need to handle any other cases -- just continue if the count is less than the max
```